File: Day 4/49. Group Anagrams.ts
function groupAnagrams(strs: string[]): string[][] {
    let obj = {};
    for (let words of strs){
        let letters = words.split("").sort().join("");
        obj[letters] ? obj[letters].push(words) : obj[letters] = [words];
    }
    return Object.values(obj);
};

File: Day 4/49. Group Anagrams.py
# Solution Link https://leetcode.com/problems/group-anagrams/submissions/867584244/

class Solution:
    def groupAnagrams(self, strs):
        mapResult = {}
        
        if len(strs) == 0 :
            return [['']]
        
        for i in strs:
            res =str(''.join(sorted(i)))
            if res in mapResult:
                mapResult[res].append(str(i))
            else:
                mapResult[res] = [str(i)]

        return list(mapResult.values())



Solution().groupAnagrams(["eat", "tea", "tan", "ate", "nat", "bat"])